### DISCUSSION TITLE
Codespaces settings.json location

### DISCUSSION THREAD

**Discussion by sobolevn** [community/community#27142] [Codespaces]
<p>I am trying to replace <code>settings.json</code> file in a Codespace by my own version.</p>
<p>But, the problem is that I cannot find native <code>settings.json</code> file. It should be located under <code>$HOME/.config/Code/User/settings.json</code>. But, <code>/$HOME/.config/</code> does not exist.</p>
<p>Instead, <code>vscode</code> shows that <code>settings.json</code> is located under <code>/User/settings.json</code>.<br>
Screenshot:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/b/2bc77c48e7b715999c36d9d1df5e56a713e0de3a.png" data-download-href="/uploads/short-url/6fhVt3NVWOSgvIn3gzGBMlNDuoW.png?dl=1" title="Снимок экрана 2021-01-10 в 20.22.56" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/4660275/181102040-d4cacafd-8f55-4065-93f3-84eccfbc4ef8.png" alt="Снимок экрана 2021-01-10 в 20.22.56" data-base62-sha1="6fhVt3NVWOSgvIn3gzGBMlNDuoW" width="690" height="431" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2bc77c48e7b715999c36d9d1df5e56a713e0de3a_2_690x431.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2bc77c48e7b715999c36d9d1df5e56a713e0de3a_2_1035x646.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2bc77c48e7b715999c36d9d1df5e56a713e0de3a_2_1380x862.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2bc77c48e7b715999c36d9d1df5e56a713e0de3a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Снимок экрана 2021-01-10 в 20.22.56</span><span class="informations">2560×1600 257 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But, this folder does not exist on disk.</p>
<p>What should I do?</p>

**Comment by legomushroom** [comment]
<p>Hi <a class="mention" href="/sobolevn">@sobolevn</a>,</p>
<p>There are multiple different settings file you can use to achieve this, the file you want to change is kept in the browser storage so won’t be directly editable in the web case of the Codespaces. You can:</p>
<ul>
<li>add the <code>.vscode/settings.json</code> file into the root of the workspace instead (<code>workspace settings</code>)</li>
<li>modify the <code>/home/codespace/.vscode-remote/data/Machine/settings.json</code> file instead (<code>remote codespaces settings</code>) - this file will be present on the codespace filesystem as you would expect.</li>
</ul>
<p>Hope it helps :slight_smile:</p>

**Comment by jkeech** [comment]
<p><a class="mention" href="/sobolevn">@sobolevn</a>, you can also leverage VS Code Settings Sync (which is enabled in Github Codespaces) to sync your user settings across codespaces and between a codespace and VS Code on the desktop.</p>

**Comment by sobolevn** [comment]
<p>Sorry, but I have checked the proposed solution and no files are present in <code>/home</code> directory.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/5/456e8a67d27b43f9b7de23f23f2e13c6efd3a611.png" data-download-href="/uploads/short-url/9UdMduHN6kZHq0cgjTt57PHMR4R.png?dl=1" title="Снимок экрана 2021-01-14 в 11.03.16" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/4660275/181102043-a5e737ba-bd6e-4aeb-b1de-c38dbcc696f5.png" alt="Снимок экрана 2021-01-14 в 11.03.16" data-base62-sha1="9UdMduHN6kZHq0cgjTt57PHMR4R" width="690" height="200" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/5/456e8a67d27b43f9b7de23f23f2e13c6efd3a611_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Снимок экрана 2021-01-14 в 11.03.16</span><span class="informations">956×278 24.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>So, <code>/home/codespace/.vscode-remote/data/Machine/settings.json</code> does not exist.</p>
<p>I have tried to create this folder manually with <code>mkdir -p /home/codespace/.vscode-remote/data/Machine/</code> and then changing something inside the <code>Remote</code> settings. Still no <code>settings.json</code> file.</p>

**Comment by sobolevn** [comment]
<p>The <code>Remote</code> settings file is in <code>/root/.vscode-remote/data/Machine/settings.json</code></p>

**Comment by jkeech** [comment, marked as answer]
<p>To clarify, the settings file will be in <code>~/.vscode-remote/data/Machine/settings.json</code>, and the exact value of <code>~</code> or <code>$HOME</code> will depend on the container image that is being used for the codespace. The default codespaces image runs as the <code>codespace</code> user with a home directory of <code>/home/codespace</code>, but a custom image might use <code>root</code> with a home directory of <code>/root</code>, or any other user and home directory path.</p>

**Reply by Musicminion** [reply]
So, what about devcontainer? I use github codespace & devcontainer, however, I can't find setting.json in
- `~/.vscode-remote/data/Machine/settings.json` or
-  `/home/codespace`

I just want to set vscode-extensiton config file `setting.json` when codespace is created, is there any way?

**Reply by virzak** [reply]
Not sure this should be marked as answer. I have no idea where to copy settings.json file to.
